core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -2704,28 +2710,30 @@ class ReplicaManagerTest {
       time = time,
       scheduler = time.scheduler,
       logManager = logManager,
-      quotaManagers = QuotaFactory.instantiate(config, metrics, time, ""),
+      quotaManagers = quotaManager,
       metadataCache = MetadataCache.zkMetadataCache(config.brokerId, config.interBrokerProtocolVersion),
       logDirFailureChannel = new LogDirFailureChannel(config.logDirs.size),
       alterPartitionManager = alterPartitionManager,
       threadNamePrefix = Option(this.getClass.getName))
 
-    logManager.startup(Set.empty[String])
-
-    // Create a hosted topic, a hosted topic that will become stray, and a stray topic
-    val validLogs = createHostedLogs("hosted-topic", numLogs = 2, replicaManager).toSet
-    createHostedLogs("hosted-stray", numLogs = 10, replicaManager).toSet
-    createStrayLogs(10, logManager)
+    try {
+      logManager.startup(Set.empty[String])
 
-    val allReplicasFromLISR = Set(new TopicPartition("hosted-topic", 0), new TopicPartition("hosted-topic", 1))
+      // Create a hosted topic, a hosted topic that will become stray, and a stray topic
+      val validLogs = createHostedLogs("hosted-topic", numLogs = 2, replicaManager).toSet
+      createHostedLogs("hosted-stray", numLogs = 10, replicaManager).toSet
+      createStrayLogs(10, logManager)
 
-    replicaManager.updateStrayLogs(replicaManager.findStrayPartitionsFromLeaderAndIsr(allReplicasFromLISR))
+      val allReplicasFromLISR = Set(new TopicPartition("hosted-topic", 0), new TopicPartition("hosted-topic", 1))
 
-    assertEquals(validLogs, logManager.allLogs.toSet)
-    assertEquals(validLogs.size, replicaManager.partitionCount.value)
+      replicaManager.updateStrayLogs(replicaManager.findStrayPartitionsFromLeaderAndIsr(allReplicasFromLISR))
 
-    replicaManager.shutdown()
-    logManager.shutdown()
+      assertEquals(validLogs, logManager.allLogs.toSet)
+      assertEquals(validLogs.size, replicaManager.partitionCount.value)
+    } finally {
+      replicaManager.shutdown()
+      logManager.shutdown()
```

Comment:
Can you close both of them safely like below ignoring any intermediate failures?

```
      CoreUtils.swallow(replicaManager.shutdown(checkpointHW = false), this)
      CoreUtils.swallow(remoteLogManager.close(), this)

```

Reply:
Do we want to close them quietly? This will suppress memory leaks since we won't get to know if the close is not happening correctly. It's better to fail when close has an error so that we get to know that something is wrong with tests,

Reply:
I think what Satish wants is we close both of them without throwing any exception to not closing the 2nd instance.

Reply:
So, I think I should only swallow in more than 1 instances to get close in finally block. Is that what you expect @satishd ?

Reply:
That is fair. In that case, we should perhaps use `public static void closeQuietly(AutoCloseable closeable, String name, AtomicReference<Throwable> firstException)` and end the finally with something similar to

```
if (firstException != null) {
           throw new KafkaException("Failed to close", exception);
        }
```

Reply:
Right @showuon , what I meant here was try closing the second one even if the earlier invocation has thrown an error to avoid leaks caused by not closing the second instance. We should not change any of the other places to swallow the exceptions where there is only one single invocation of shutting down of replicamanager or logmanager. 

One way to do that is what @divijvaidya suggested.

Reply:
I've updated the PR. Thanks.